rosidl_runtime_cpp/include/rosidl_runtime_cpp/traits.hpp

```diff
@@ -15,11 +15,137 @@
 #ifndef ROSIDL_RUNTIME_CPP__TRAITS_HPP_
 #define ROSIDL_RUNTIME_CPP__TRAITS_HPP_
 
+#include <codecvt>
+#include <string>
 #include <type_traits>
 
 namespace rosidl_generator_traits
 {
 
+void value_to_yaml(bool value, std::ostream & out)
+{
+  out << (value ? "true" : "false");
+}
+
+void character_value_to_yaml(unsigned char value, std::ostream & out)
+{
+  auto flags = out.flags();
+  out << "0x" << std::hex << std::setw(2) << std::setfill('0') << +value;
+  out.flags(flags);
+}
+
+void character_value_to_yaml(char16_t value, std::ostream & out)
+{
+  auto flags = out.flags();
+  out << "\"\\u" << std::hex << std::setw(4) << std::setfill('0') << \
+    *reinterpret_cast<const int16_t *>(&value) << "\"";
+  out.flags(flags);
+}
+
+void value_to_yaml(float value, std::ostream & out)
+{
+  auto flags = out.flags();
+  out << std::showpoint << value;
+  out.flags(flags);
+}
+
+void value_to_yaml(double value, std::ostream & out)
+{
+  auto flags = out.flags();
+  out << std::showpoint << value;
+  out.flags(flags);
+}
+
+void value_to_yaml(long double value, std::ostream & out)
+{
+  auto flags = out.flags();
+  out << std::showpoint << value;
+  out.flags(flags);
+}
+
+void value_to_yaml(uint8_t value, std::ostream & out)
+{
+  out << +value;
+}
+
+void value_to_yaml(int8_t value, std::ostream & out)
+{
+  out << +value;
+}
+
+void value_to_yaml(uint16_t value, std::ostream & out)
+{
+  out << value;
+}
+
+void value_to_yaml(int16_t value, std::ostream & out)
+{
+  out << value;
+}
+
+void value_to_yaml(uint32_t value, std::ostream & out)
+{
+  out << value;
+}
+
+void value_to_yaml(int32_t value, std::ostream & out)
+{
+  out << value;
+}
+
+void value_to_yaml(uint64_t value, std::ostream & out)
+{
+  out << value;
+}
+
+void value_to_yaml(int64_t value, std::ostream & out)
+{
+  out << value;
+}
+
+void value_to_yaml(const std::string & value, std::ostream & out)
+{
+  out << "\"";
+  size_t index = 0;
+  while (index < value.size()) {
+    size_t pos = value.find_first_of("\\\"", index);
+    if (pos == std::string::npos) {
+      pos = value.size();
+    }
+    out.write(&value[index], pos - index);
+    if (pos >= value.size()) {
+      break;
+    }
+    out << "\\" << value[pos];
+    index = pos + 1;
+  }
+  out << "\"";
+}
+
+void value_to_yaml(const std::u16string & value, std::ostream & out)
+{
+  out << "\"";
+  std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> convert;
+  auto flags = out.flags();
+  size_t index = 0;
+  while (index < value.size()) {
+    int16_t character = *reinterpret_cast<const int16_t *>(&value[index]);
```

Comment:
same about using `static_cast` instead

Reply:
Done in c71a01c401d9b78cc563c86844dc2692c2953832.